modules/apps/27-interchain-accounts/controller/keeper/handshake.go

```diff
@@ -32,26 +34,25 @@ func (k Keeper) OnChanOpenInit(
 		return sdkerrors.Wrapf(channeltypes.ErrInvalidChannelOrdering, "expected %s channel, got %s", channeltypes.ORDERED, order)
 	}
 
-	connSequence, err := icatypes.ParseControllerConnSequence(portID)
-	if err != nil {
-		return sdkerrors.Wrapf(err, "expected format %s, got %s", icatypes.ControllerPortFormat, portID)
+	if !strings.HasPrefix(portID, icatypes.PortPrefix) {
+		return sdkerrors.Wrapf(icatypes.ErrInvalidControllerPort, "expected %s{owner-account-address}, got %s", icatypes.PortPrefix, portID)
 	}
 
-	counterpartyConnSequence, err := icatypes.ParseHostConnSequence(portID)
-	if err != nil {
-		return sdkerrors.Wrapf(err, "expected format %s, got %s", icatypes.ControllerPortFormat, portID)
+	if counterparty.PortId != icatypes.PortID {
+		return sdkerrors.Wrapf(icatypes.ErrInvalidHostPort, "expected %s, got %s", icatypes.PortID, counterparty.PortId)
 	}
 
-	if err := k.validateControllerPortParams(ctx, connectionHops, connSequence, counterpartyConnSequence); err != nil {
-		return sdkerrors.Wrapf(err, "failed to validate controller port %s", portID)
+	var metadata icatypes.Metadata
+	if err := icatypes.ModuleCdc.UnmarshalJSON([]byte(version), &metadata); err != nil {
+		return sdkerrors.Wrapf(icatypes.ErrUnknownDataType, "cannot unmarshal ICS-27 interchain accounts metadata")
 	}
 
-	if counterparty.PortId != icatypes.PortID {
-		return sdkerrors.Wrapf(porttypes.ErrInvalidPort, "expected %s, got %s", icatypes.PortID, counterparty.PortId)
+	if err := k.validateConnectionParams(ctx, connectionHops, metadata.ControllerConnectionId, metadata.HostConnectionId); err != nil {
+		return err
 	}
 
-	if version != icatypes.VersionPrefix {
-		return sdkerrors.Wrapf(icatypes.ErrInvalidVersion, "expected %s, got %s", icatypes.VersionPrefix, version)
+	if metadata.Version != icatypes.Version {
```

Comment:
Do we want to allow relayers to pass in empty string for version here? Or is this a future improvement?

ref: https://github.com/cosmos/ibc/pull/629/files

cc: @colin-axner 

This can also be implemented in a separate PR

Reply:
Am I missing something here? `OnChanOpenInit` will never be invoked by relayers directly, right? 
Otherwise, you'll potentially end up in a situation with an open channel with [no port bound/capability claimed](https://github.com/cosmos/ibc-go/blob/main/modules/apps/27-interchain-accounts/controller/keeper/account.go#L24-L31)?

We should probably be adding a defensive check to `OnChanOpenInit` to error if a controller port is not bound.

Reply:
`OnChanOpenInit` can be invoked directly by relayers in the case where the port is bound (InitInterchainAccount has been called) + a channel is not the active channel. If a channel closes, a relayer will be able to create a new channel on bound port. 

> We should probably be adding a defensive check to OnChanOpenInit to error if a controller port is not bound.

As far as I know, this is handled by core IBC already. You can't open a channel on a port that is not bound already.

Reply:
You're right, it would be handled by core [here](https://github.com/cosmos/ibc-go/blob/main/modules/core/keeper/msg_server.go#L264) in the case where a port is not bound 👍